packages/sit-onyx/src/components/OnyxDataGrid/features/sorting/SortAction.vue

```diff
@@ -0,0 +1,41 @@
+<script setup lang="ts">
+import arrowSmallDown from "@sit-onyx/icons/arrow-small-down.svg?raw";
+import arrowSmallUp from "@sit-onyx/icons/arrow-small-up.svg?raw";
+import arrowsSort from "@sit-onyx/icons/arrows-sort.svg?raw";
+import { computed } from "vue";
+import { OnyxIconButton } from "../../../..";
+import { injectI18n } from "../../../../i18n";
+import type { SortDirection } from "./sorting";
+import { nextSortDirection } from "./sorting";
+
+const props = defineProps<{
+  /**
+   * Label of this Column.
+   */
+  columnLabel: string;
+  /**
+   * The current sorting direction, that should be indicated.
+   */
+  sortDirection?: SortDirection;
+}>();
+
+const { t } = injectI18n();
+
+const icon = computed(() =>
+  props.sortDirection === "asc"
+    ? arrowSmallUp
+    : props.sortDirection === "desc"
+      ? arrowSmallDown
+      : arrowsSort,
+);
+
+const buttonLabel = computed(() => {
+  const nextDirection = nextSortDirection(props.sortDirection);
+  return t.value(`data-grid.head.sorting-action-${nextDirection}`, {
+    field: props.columnLabel,
+  });
+});
+</script>
+<template>
+  <OnyxIconButton color="neutral" :icon="icon" size="12px" :label="buttonLabel" />
```

Comment:
If no sort order is set, the icon should only be shown on hover: https://www.figma.com/design/YfEUBOHk4J4nYrk04geswG/Onyx-Component-Library?node-id=4435-44766&m=dev

Reply:
out-of-scope for now